language/move-lang/tests/move_check/unit_test/multiple_errors.move

```diff
@@ -0,0 +1,45 @@
+// test that we handle multiple errors in the same file correctly and don't stop at the first one
+address 0x1 {
+module M {
+    #[test_only]
+    struct Foo {}
+
+    public fun foo() { }
+
+    #[test_only]
+    public fun bar() { }
+
+    #[test]
+    public fun go() { }
+
+    #[test(_a=0x1)]
+    public fun a(_a: signer) { }
+
+    // failure: invalid value in test parameter assignment
+    #[test(_a=Foo)]
+    public fun b(_a: signer) { }
+
+    #[test(_a=0x1, _b=0x2)]
+    public fun c(_a: signer, _b: signer) { }
+
+    // failure: double annotation
+    #[test(_a=0x1)]
+    #[test(_b=0x2)]
+    public fun d(_a: signer, _b: signer) { }
+
+    // failure: annotated as both test and test_only
+    #[test(_a=0x1)]
```

Comment:
Just wondering: have you decided to not provide a `Test::create_signer` test-only function but instead use attributes for this? I think this is a mistake and will restrict the framework unnecessarily, as you can't have a dynamic number of signers. (There are a number of use cases I can imagine for this.). Also I would like to see parameters to tests reserved for parameterized testing, not for the special case of signers. This is also what people expect when seeing a unit test.

I recognize I should have like flagged this earlier in the design docs, so it might be too late to change. Perhaps later we can still provide the create_signer function and ignore this _a = 0x1 stuff...

Reply:
I haven't added test-only natives yet (but am in the process of implementing this), so if we want a `Test::create_signer` function we could add it then (and removing these signer parameters wouldn't be too difficult).

The main reason I went with parameters for the signers, was a number of people on the team had serious concerns about having a signer creation function, even in test-only code. But I'll bring it up for a discussion at tomorrow's meeting, and if folks are OK with it or if we can find a good way to mitigate concerns around it, I can most definitely do that instead of this without too much issue (but, in a subsequent PR).

Reply:
I would definitely be all for a signer creation function for natives. 

But I agree that at least for now allowing signers to be created this way is the easiest way for this thing to go in and start being useful. The invocation APIs for execution_function in the VM make it easy to make signers in this way, so it makes sense to me that the tests are currently parameterized like this